tests/integration/targets/azure_rm_afdendpoint/tasks/main.yml

```
@@ -0,0 +1,169 @@
+
+- name: Create Standard Frontdoor Profile
+  azure_rm_cdnprofile:
+    name: "{{ profile }}"
+    location: "Global"
+    resource_group: "{{ resource_group }}"
+    sku: "standard_azurefrontdoor"
+    state: "present"
+
+- name: Create EndPoint
+  azure_rm_afdendpoint:
+    name: "{{ endpoint }}"
+    profile_name: "{{ profile }}"
+    resource_group: "{{ resource_group }}"
+    enabled_state: "Enabled"
+    auto_generated_domain_name_label_scope: "TenantReuse"
+    tags:
+      test: "tag value"
+    state: "present"
+  register: output
+
+- name: Assert the resource was created and returns expected variables
+  ansible.builtin.assert:
+    that:
+      - output.changed
+      - output.id
+      - output.host_name
+
+- name: Update EndPoint (idempotent test)
+  azure_rm_afdendpoint:
+    name: "{{ endpoint }}"
+    profile_name: "{{ profile }}"
+    resource_group: "{{ resource_group }}"
+    enabled_state: "Enabled"
+    auto_generated_domain_name_label_scope: "TenantReuse"
+    tags:
+      test: "tag value"
+    state: "present"
+  register: output
+
+- name: Assert the resource has not changed
+  ansible.builtin.assert:
+    that:
+      - not output.changed
+
+- name: Load Endpoint info
+  azure_rm_afdendpoint_info:
+    name: "{{ endpoint }}"
+    profile_name: "{{ profile }}"
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- name: Assert the resource has the proper values set
+  ansible.builtin.assert:
+    that:
+      - output.afdendpoints | length == 1
+      - output.afdendpoints[0].auto_generated_domain_name_label_scope == 'TenantReuse'
+      - output.afdendpoints[0].enabled_state == 'Enabled'
+      - output.afdendpoints[0].host_name
+      - output.afdendpoints[0].id
+      - output.afdendpoints[0].profile_name == profile
+      - output.afdendpoints[0].tags.test == 'tag value'
+      - not output.changed
+
+- name: Update EndPoint (with different values)
+  azure_rm_afdendpoint:
+    name: "{{ endpoint }}"
+    profile_name: "{{ profile }}"
+    resource_group: "{{ resource_group }}"
+    enabled_state: "Disabled"
+    tags:
+      test: "new tag value"
+      newtag: "another value"
+    state: "present"
+  register: output
+
+- name: Assert the resource has changed
+  ansible.builtin.assert:
+    that:
+      - output.changed
+
+- name: Update EndPoint (idempotent test)
+  azure_rm_afdendpoint:
+    name: "{{ endpoint }}"
+    profile_name: "{{ profile }}"
+    resource_group: "{{ resource_group }}"
+    enabled_state: "Disabled"
+    tags:
+      test: "new tag value"
+      newtag: "another value"
+    state: "present"
+  register: output
+
+- name: Assert the resource has not changed
+  ansible.builtin.assert:
+    that:
+      - not output.changed
+
+- name: Update EndPoint (replace tags)
+  azure_rm_afdendpoint:
+    name: "{{ endpoint }}"
+    profile_name: "{{ profile }}"
+    resource_group: "{{ resource_group }}"
+    enabled_state: "Disabled"
+    append_tags: False
```

Truthy value should be one of
```suggestion
    append_tags: false
```